### ISSUE TITLE
[Epic] Why are Form526 Submissions Invalid

### ISSUE THREAD

**Issue by annaswims** [department-of-veterans-affairs/va.gov-team#15310]
7% of requests  to v0/disability_compensation_forms/submit_all_claim  had an API response that the data was invalid.
This shouldn't be happening because the FE should be validating. Figure out wha the most common validation errors are and whether the FE should be more strict or the BE should be less strict.

[sentry issue](http://sentry.vfs.va.gov/organizations/vsp/issues/417/?environment=production&project=3&statsPeriod=14d)

![Screen Shot 2020-10-26 at 3.23.55 PM.png](https://images.zenhubusercontent.com/5bb3c3a61c43df49f8b08cfb/9ddfb91d-622d-4422-9a27-18aec90022f6)

**Comment by annaswims** [comment]
It's looking like these two are the most common.
```` 
[
{
code: 100, 
detail: form - The property '#/newPrimaryDisabilities/0' did not contain a required property of 'cause' in schema 283df042-731e-5700-9d76-c7d95c3f3ac4#, 
source: 
{
pointer: data/attributes/form
}
, 
status: 422, 
title: Form The property '#/newPrimaryDisabilities/0' did not contain a required property of 'cause' in schema 283df042-731e-5700-9d76-c7d95c3f3ac4#
}
]
````

````
[
{
code: 100, 
detail: form - The property '#/vaTreatmentFacilities/0/treatedDisabilityNames' did not contain a minimum number of items 1 in schema 283df042-731e-5700-9d76-c7d95c3f3ac4, 
source: 
{
pointer: data/attributes/form
}
, 
status: 422, 
title: Form The property '#/vaTreatmentFacilities/0/treatedDisabilityNames' did not contain a minimum number of items 1 in schema 283df042-731e-5700-9d76-c7d95c3f3ac4
}
]
````

**Comment by annaswims** [comment]
we validate against https://github.com/department-of-veterans-affairs/vets-json-schema/blob/master/dist/21-526EZ-ALLCLAIMS-schema.json

**Comment by annaswims** [comment]
[this error](http://sentry.vfs.va.gov/organizations/vsp/issues/417/events/bd3855dcc0824571909e9c887db8d1dc/?environment=production&project=3&statsPeriod=14d) corresponds with this data.

_puts InProgressForm.find(2272307).form_data_

````  
"va_treatment_facilities": [
    {
      "treatment_center_name": "James A Lovell",
      "treatment_date_range": {
        "from": "2015-02-XX"
      },
      "treatment_center_address": {
        "country": "USA",
        "city": "North Chicago",
        "state": "IL"
      },
      "treated_disability_names": {
        "condition 1, another condition, more, something, another medical thing": true
      }
    }
  ],
````

the value of treated_disability_names should be an array of strings, but is not.  I redacted the disability names, but you get the point.

**Comment by annaswims** [comment]
[this error](http://sentry.vfs.va.gov/organizations/vsp/issues/417/events/17de1a3d53a94615852cec8cb3c32f82/?environment=production&project=3&statsPeriod=14d)
from this data 
`puts InProgressForm.where(user_uuid: "2d3ef1e53c6d46629ee52fb35cafaee3").first.form_data`
looks like the last element in the array is probably not valid
````
        "persons_involved": [
            {
                "description": "SGT\"REDACTED\" REDACTED",
                "injury_death": "killedInAction",
                "injury_death_date": "2005-05-08",
                "name": {
                    "first": "REDACTED",
                    "last": "REDACTED ",
                    "middle": "REDACTED"
                },
                "rank": "SGT",
                "unit_assigned": "REDACTED",
                "view:individual_add_msg": {},
                "view:service_member": true
            },
            {
                "description": "REDACTED",
                "injury_death": "woundedInAction",
                "injury_death_date": "2004-05-08",
                "name": {
                    "last": "REDACTED"
                },
                "rank": "SGT",
                "unit_assigned": "REDACTED",
                "view:individual_add_msg": {},
                "view:service_member": true
            },
            {
                "name": {},
                "view:individual_add_msg": {}
            }
        ],
````

````
[
{
code: 100, 
detail: form - The property '#/form0781/incidents/0/personsInvolved' of type object did not match the following type: array in schema 283df042-731e-5700-9d76-c7d95c3f3ac4, 
source: 
{
pointer: data/attributes/form
}
, 
status: 422, 
title: Form The property '#/form0781/incidents/0/personsInvolved' of type object did not match the following type: array in schema 283df042-731e-5700-9d76-c7d95c3f3ac4
}
]
````

**Comment by luke-gcio** [comment]
Added some points and moved to sprint 34 - we (FE team and Luke) are meeting on Thursday to do some more investigation.

**Comment by luke-gcio** [comment]
Pausing this - Anna will wait until the existing fixes are made then revisit.

**Comment by edmangimelli** [comment]
talked about this in back-end backlog grooming today.

we've decided that we shouldn't mess with this until the SIP bug fixes have been merged

**Comment by annaswims** [comment]
Actually, I take back what I said in grooming. https://github.com/department-of-veterans-affairs/va.gov-team/issues/15368 explains nearly all of the validation errors. We stopped seeing the validation errors after https://github.com/department-of-veterans-affairs/vets-website/pull/15179 was merged. The problem isn't fixed, but #15179 is a better description of what's happening.
<img width="1531" alt="Screen Shot 2020-12-29 at 3 19 27 PM" src="https://user-images.githubusercontent.com/19188/103311800-4f34cc80-49e9-11eb-8cae-01c8bae4f08e.png">
